src/aal/aal_mips.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#define SNMALLOC_VA_BITS_64
```

Comment:
We probably need to guard this on a `__mips64` or similar define.  Looking at the rest of the PAL, there's no reason that it couldn't work with the n32 or o32 ABIs though, so maybe we should define it conditionally on those.  The `__mips_n64`, `__mips_n32`, and `__mips_o32` macros can be used to differentiate on these.

Reply:
Those are FreeBSD-specific (Clang will define those all the time, but GCC doesn't for Linux); the correct preprocessor checks for MIPS are checking `_MIPS_SIM` against `_ABIO32`/`_ABIN32`/`_ABI64`.